apps/challenges/aws_utils.py

```diff
@@ -1061,6 +1061,257 @@ def scale_resources(challenge, worker_cpu_cores, worker_memory):
+                )
+            ec2.delete_network_interface(
+                NetworkInterfaceId=eni["NetworkInterfaceId"]
+            )
+
+        for lb in elb.describe_load_balancers()["LoadBalancerDescriptions"]:
+            if lb["VPCId"] == vpc_id:
+                elb.delete_load_balancer(
+                    LoadBalancerName=lb["LoadBalancerName"]
+                )
+
+        for lb in elbv2.describe_load_balancers()["LoadBalancers"]:
+            if lb["VpcId"] == vpc_id:
+                elbv2.delete_load_balancer(
+                    LoadBalancerArn=lb["LoadBalancerArn"]
+                )
+
+        for sg in ec2_vpc.security_groups.all():
+            if sg.group_name != "default":
+                if sg.ip_permissions:
+                    sg.revoke_ingress(IpPermissions=sg.ip_permissions)
+                if sg.ip_permissions_egress:
+                    sg.revoke_egress(IpPermissions=sg.ip_permissions_egress)
+                sg.delete()
+
+        for subnet in ec2_vpc.subnets.all():
+            subnet.delete()
+
+        for rt in ec2_vpc.route_tables.all():
+            if not rt.associations:  # Ignore the main route table
+                rt.delete()
+            else:
+                for assoc in rt.associations:
+                    if (
+                        not assoc.main
+                    ):  # Ignore the main route table association
+                        assoc.delete()
+
+        for acl in ec2_vpc.network_acls.all():
+            if not acl.is_default:
+                acl.delete()
+
+        for ep in ec2.describe_vpc_endpoints(
+            Filters=[{"Name": "vpc-id", "Values": [vpc_id]}]
+        )["VpcEndpoints"]:
+            logger.info(f'Deleting VPC endpoint: {ep["VpcEndpointId"]}')
+            ec2.delete_vpc_endpoints(VpcEndpointIds=[ep["VpcEndpointId"]])
+
+        ec2_vpc.delete()
+
+    except ClientError as e:
+        logger.exception(e)
+
+
+@app.task
+def delete_code_upload_infrastructure(challenge):
+    """
+    Delete the infrastructure associated with a challenge that uses code upload.
+    - Delete EKS Cluster and Nodegroup
+    - Delete EFS
+    - Delete VPC
+    - Delete IAM Roles
+    """
+
+    from .models import ChallengeEvaluationCluster
+    from .utils import get_aws_credentials_for_challenge
+
+    for obj in serializers.deserialize("json", challenge):
+        challenge_obj = obj.object
+
+    challenge_evaluation_cluster = ChallengeEvaluationCluster.objects.get(
+        challenge=challenge_obj
+    )
+
+    challenge_aws_keys = get_aws_credentials_for_challenge(challenge.pk)
+    iam_client = get_boto3_client("iam", challenge_aws_keys)
+    eks_client = get_boto3_client("eks", challenge_aws_keys)
+    efs_client = get_boto3_client("efs", challenge_aws_keys)
+    ec2_client = get_boto3_client("ec2", challenge_aws_keys)
+    elb_client = get_boto3_client("elb", challenge_aws_keys)
+    elbv2_client = get_boto3_client("elbv2", challenge_aws_keys)
+
```

Comment:
This is incorrect. This is now how we do things in the AWS utils during EKS cluster creation. Recommend you to read.